vue/components/actions/action-handler-mixin.vue

```diff
@@ -54,23 +72,41 @@ export default {
         return false;
       }
 
-      const handlers = getHandlersWithPayload(this);
+      let lastExecutedHandler;
+      const handlers = [
+        ...getHandlersWithPayload(this, actionDFName),
+        ...getHandlersWithPayload(this, 'processActionsGeneric'),
+      ];
+
       if (!await asyncSome(
         handlers,
-        async (handler) => (handler.handler[actionDFName](action, handler.payload, extraData)),
+        async (handler) => {
+          lastExecutedHandler = handler;
+          return (handler.handler[actionDFName] ??
+            handler.handler.processActionsGeneric)(action, handler.payload, extraData);
+        },
       )) {
-        const actualActionName = actionDFName;
-        actionDFName = 'processActionsGeneric';
-        const hndlrs = getHandlersWithPayload(this);
-        if (!await asyncSome(
-          hndlrs,
-          async (handler) => (handler.handler[actionDFName](action, handler.payload, extraData)),
-        )) {
-          console.warn(
-            `[unprocessed] Action ${this.$options.name}.${actualActionName}()`,
-          );
+        // no handler was executed
+        if (lastExecutedHandler) {
+          // means we know some handlers were executed, return first one
+          lastExecutedHandler = handlers.shift();
         }
+        // if there were no executions, lastExecutedHandler evaluates into undefined
+        const emitData = [
+          'action-executed',
+          { action, handler: lastExecutedHandler?.payload, extraData, actionHandled: false },
+        ];
+
+        emitEvent(this, emitData);
+        return;
       }
+      // handler was executed
+      const emitData = [
```

Comment:
tale emitData in $emit sam po sebi bi se dalo DRY-at, saj je med njima edina razlika v actionHandled (boolean), ki se nastavi v if / else if (asyncSome)